java/fury-core/src/main/java/org/apache/fury/resolver/ClassResolver.java

```diff
@@ -156,55 +157,14 @@ public class ClassResolver {
-  public static final short PRIMITIVE_LONG_CLASS_ID = (short) (PRIMITIVE_VOID_CLASS_ID + 7);
-  public static final short PRIMITIVE_DOUBLE_CLASS_ID = (short) (PRIMITIVE_VOID_CLASS_ID + 8);
-  public static final short VOID_CLASS_ID = (short) (PRIMITIVE_DOUBLE_CLASS_ID + 1);
-  public static final short BOOLEAN_CLASS_ID = (short) (VOID_CLASS_ID + 1);
-  public static final short BYTE_CLASS_ID = (short) (VOID_CLASS_ID + 2);
-  public static final short CHAR_CLASS_ID = (short) (VOID_CLASS_ID + 3);
-  public static final short SHORT_CLASS_ID = (short) (VOID_CLASS_ID + 4);
-  public static final short INTEGER_CLASS_ID = (short) (VOID_CLASS_ID + 5);
-  public static final short FLOAT_CLASS_ID = (short) (VOID_CLASS_ID + 6);
-  public static final short LONG_CLASS_ID = (short) (VOID_CLASS_ID + 7);
-  public static final short DOUBLE_CLASS_ID = (short) (VOID_CLASS_ID + 8);
-  public static final short STRING_CLASS_ID = (short) (VOID_CLASS_ID + 9);
-  public static final short PRIMITIVE_BOOLEAN_ARRAY_CLASS_ID = (short) (STRING_CLASS_ID + 1);
-  public static final short PRIMITIVE_BYTE_ARRAY_CLASS_ID = (short) (STRING_CLASS_ID + 2);
-  public static final short PRIMITIVE_CHAR_ARRAY_CLASS_ID = (short) (STRING_CLASS_ID + 3);
-  public static final short PRIMITIVE_SHORT_ARRAY_CLASS_ID = (short) (STRING_CLASS_ID + 4);
-  public static final short PRIMITIVE_INT_ARRAY_CLASS_ID = (short) (STRING_CLASS_ID + 5);
-  public static final short PRIMITIVE_FLOAT_ARRAY_CLASS_ID = (short) (STRING_CLASS_ID + 6);
-  public static final short PRIMITIVE_LONG_ARRAY_CLASS_ID = (short) (STRING_CLASS_ID + 7);
-  public static final short PRIMITIVE_DOUBLE_ARRAY_CLASS_ID = (short) (STRING_CLASS_ID + 8);
-  public static final short STRING_ARRAY_CLASS_ID = (short) (PRIMITIVE_DOUBLE_ARRAY_CLASS_ID + 1);
-  public static final short OBJECT_ARRAY_CLASS_ID = (short) (PRIMITIVE_DOUBLE_ARRAY_CLASS_ID + 2);
-  public static final short ARRAYLIST_CLASS_ID = (short) (PRIMITIVE_DOUBLE_ARRAY_CLASS_ID + 3);
-  public static final short HASHMAP_CLASS_ID = (short) (PRIMITIVE_DOUBLE_ARRAY_CLASS_ID + 4);
-  public static final short HASHSET_CLASS_ID = (short) (PRIMITIVE_DOUBLE_ARRAY_CLASS_ID + 5);
-  public static final short CLASS_CLASS_ID = (short) (PRIMITIVE_DOUBLE_ARRAY_CLASS_ID + 6);
-  public static final short EMPTY_OBJECT_ID = (short) (PRIMITIVE_DOUBLE_ARRAY_CLASS_ID + 7);
   private static final int initialCapacity = 128;
   // use a lower load factor to minimize hash collision
   private static final float loadFactor = 0.25f;
   private static final float furyMapLoadFactor = 0.25f;
   private static final String META_SHARE_FIELDS_INFO_KEY = "shareFieldsInfo";
   private static final ClassInfo NIL_CLASS_INFO =
-      new ClassInfo(null, null, null, null, false, null, null, ClassResolver.NO_CLASS_ID);
-
+      new ClassInfo(null, null, null, null, false, null, null, BuiltinClassId.NO_CLASS_ID);
   private final Fury fury;
-  private ClassInfo[] registeredId2ClassInfo = new ClassInfo[] {};
```

Comment:
`registeredId2ClassInfo` is in the critical path, we'd better to keep it here instead of `extRegistry`, which incur another pointer indirection.